include/odp/api/spec/crypto.h

```diff
@@ -510,6 +510,15 @@ typedef struct odp_crypto_session_param_t {
 	 */
 	odp_pool_t output_pool;
 
+	/** Bitstring mode
+	 *
+	 *  Cipher and auth will operate in bitstring mode. Fields cipher_range
+	 *  and auth_range will count bits for offset and length, rather than
+	 *  full bytes. Note: not algorithms support this mode of operation, it
+	 *  has to be reported via bitstring field in cipher and auth
+	 *  capability. */
+	odp_bool_t bitstring;
```

Comment:
The term `bitstring` is a bit generic as there are lots of potential uses of bit strings. Perhaps call this `bitstring_mode`, which also has the advantage of matching the documentation here? Same comment for other two uses of this field name.

Reply:
Typo: "Note: not algorithms.." should be "Note: not all algo ..."

Reply:
I think it would be simpler API if we just divide algorithms into byte and bit offset/length ones. Kasumi/Snow/ZUC would expect that cipher_ and auth_range fields contain bit offset/length. Others expect bytes. So, all bitstring specification can be removed and replaced with something like this:

	/** Data range to apply cipher
+	 *   Range is specified in bits for Kasumi/Snow/ZUC and in bytes for all other algorithms. 
+	*/
	odp_packet_data_range_t cipher_range;

	/** Data range to authenticate
+	 *   Range is specified in bits for Kasumi/Snow/ZUC and in bytes for all other algorithms. 
+	*/
	odp_packet_data_range_t auth_range;

} odp_crypto_packet_op_param_t;

Reply:
When bit offsets are used do we have to be concerned about endianness?

Reply:
@psavol I thought about this (and it is how DPDK defines those algorithms). However 128-EEA2/128-EIA2 algorithms are plain AES-CTR/AES-CMAC algorithms, but applied to bit strings, rather than bytes.

@Bill-Fischofer-Linaro Hmm. I think no. There are no chips with different endianness of bits inside a byte.